pkg/detectors/gitlab/gitlab.go

```diff
@@ -14,21 +14,20 @@ import (
 
 type Scanner struct {
 	client *http.Client
-
 	detectors.EndpointSetter
 }
 
 // Ensure the Scanner satisfies the interfaces at compile time.
 var _ detectors.Detector = (*Scanner)(nil)
-var _ detectors.Versioner = (*Scanner)(nil)
 var _ detectors.EndpointCustomizer = (*Scanner)(nil)
+var _ detectors.Versioner = (*Scanner)(nil)
 
 func (Scanner) Version() int            { return 1 }
 func (Scanner) DefaultEndpoint() string { return "https://gitlab.com" }
 
 var (
 	defaultClient = common.SaneHttpClient()
-	keyPat        = regexp.MustCompile(detectors.PrefixRegex([]string{"gitlab"}) + `\b((?:glpat|)[a-zA-Z0-9\-=_]{20,22})\b`)
+	keyPat        = regexp.MustCompile(detectors.PrefixRegex([]string{"gitlab"}) + `\b[^glpat-]([a-zA-Z0-9\-=_]{20,22})\b`)
```

Comment:
this regex looks really suspicious. `[^glpat-]` means "exactly one character that is not a `g`, `l`, `p`, `a`, `t`, or `-`", which i'm guessing is not what you want. if you want to exclude a prefix, one solution is use something called a "negative lookbehind" but i've never done that before so you'd have to become the expert yourself. (it might also be possible to put the prefix in a capture group, make it optional, and exclude matches that contain it later, but i don't know if it's possible to do this without additionally matching the un-prefixed key, which would defeat the purpose.)

Reply:
hmm... unfortunately lookbehinds aren't supported in Go, but it probably makes more sense to explicitly filter out the results with `glpat`.